### ISSUE TITLE
Add Property "MaxValue", "MinValue" and "MouseHoveredValue" to FastNoiseLite

### ISSUE THREAD

**Issue by MatthiasBae** [godotengine/godot-proposals#11001]
### Describe the project you are working on

Working on a procedural map generator in 2D

### Describe the problem or limitation you are having in your project

No limitations but configuring noises and threshold is a little bit tedious

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Adding three properties "MaxValue", "MinValue" and "MouseHoveredValue" would help configuring noises and threshold without having to start the game and check if the set thresholds are too high / too low.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

The properties "MaxValue" and "MinValue" will change its properties value after changing any property of the noise.

The "MouseHoveredValue" will change its value when hovering above the noise preview texture in the inspector

<img width="430" alt="image" src="https://github.com/user-attachments/assets/ddeba634-8a69-4748-843f-53b240d68ed0">

### If this enhancement will not be used often, can it be worked around with a few lines of script?

You can create a script which extends from FastNoiseLite and create the values. Unfortunately the "ValidateProperty" method does not execute on every property like Frequency

<img width="582" alt="image" src="https://github.com/user-attachments/assets/86a37de0-3914-4682-bdb1-7a8cecc1ae97">

### Is there a reason why this should be core and not an add-on in the asset library?

Because FastNoiseLite is core

**Comment by Calinou** [comment]
This should not be implemented as properties in FastNoiteLite, but by modifying the inspector preview to include this information.

**Comment by MatthiasBae** [comment]
> This should not be implemented as properties in FastNoiteLite[1](#user-content-fn-1-7828d989d4226052b9e9b9388d4eeeb8), but by modifying the inspector preview to include this information.
> 
> ## Footnotes
> 1. If this is to be exposed to scripting, it should be exposed as _methods_ as these are read-only values, and only for the min/max values, not the "mouse hovered" value (which can be replaced by just getting the noise value at the cursor position). [↩](#user-content-fnref-1-7828d989d4226052b9e9b9388d4eeeb8)

You are absolutely right. Was just an example to demonstrate but I think its clear what is needed. Exposing to scripting would be useful too!
